### ISSUE TITLE
AudioChannel stop is not working

### ISSUE THREAD

**Issue by Huyston** [Kode/Kha#1155]
Minimal example: http://kodegarden.org/#5ffd9ffaa10a5fcd964f05f290619487f46207eb

**Comment by potatoelixir** [comment]
Unless I cloned something wrong, that works just fine if I get a local copy of it and run in via VSC; even with the version of Kha used by Kode Garden (which is a bit older than that of VSC). Are you using electron or running it via browser? Maybe something changed in Chromium?

**Comment by Huyston** [comment]
I tested this with Chromium and on android. Both isn't working.

**Comment by potatoelixir** [comment]
Well could you test with Electron? Electron works for me; you've said HTML5 backend doesn't work either, which it does for me, which is strange. Android isn't something I can test rn, sorry D:

**Comment by RobDangerous** [comment]
Isn't working for me in Kode Garden, tested with Firefox and Chrome. No further testing required until I had a closer look.

**Comment by potatoelixir** [comment]
Just to be clear: doesn't work for me in Kode Garden either, but works wonders in both Krom and Electron; HTML5 backend is probably outdated?

**Comment by RobDangerous** [comment]
Electron uses the HTML5 backend - also you said prevously that HTML5 works for you. Also, Kha uses a software audio mixer, so it's all the same everywhere unless it has to fall back to use audio elements when WebAudio is not working (which it also does on mobile devices for performance reasons, but we're not talking about mobile browsers here or do we?). Kode Garden's version of Kha is quite old though and maybe that's also the case for the version Huyston uses. But let's just wait until I can have a look at it myself. And then I'll also add a GitHub issue template so we get all the details right away.

**Comment by potatoelixir** [comment]
man, i literally said above i cloned the version of kha kode garden uses and it still worked fine with my vsc electron. however, i do know that my electron is super old. thus my conclusion that probably something's broken in newer browsers.

**Comment by RobDangerous** [comment]
Ah, didn't see that. So, revision c76117c, right?

**Comment by potatoelixir** [comment]
yup. not sure bout my electron(/chromium) version, but whatever is the default in vsc, from like 2018 iirc

**Comment by RobDangerous** [comment]
The default in vscode is not the version from Kode Garden.

**Comment by Huyston** [comment]
On Android the issue is a bit different, it wasn't working because of a bug in my code.
Stop works but you can't play the audio again:
Minimal example: 
Add this line in the http://kodegarden.org/#5ffd9ffaa10a5fcd964f05f290619487f46207eb:
Scheduler.addTimeTask(function () { stopSound(); }, 8);

Error is:
```2019-11-25 20:55:45.666 2099-2124/com.ktxsoftware.kha E/MediaPlayerNative: Attempt to perform seekTo in wrong state: mPlayer=0x71d44b69c0, mCurrentState=64
2019-11-25 20:55:45.666 2099-2124/com.ktxsoftware.kha E/MediaPlayerNative: error (-38, 0)
2019-11-25 20:55:45.667 2099-2099/com.ktxsoftware.kha E/MediaPlayer: Error (-38,0)
2019-11-25 20:55:45.669 2099-2124/com.ktxsoftware.kha E/MediaPlayerNative: start called in state 0, mPlayer(0x71d44b69c0)
2019-11-25 20:55:45.669 2099-2124/com.ktxsoftware.kha E/MediaPlayerNative: error (-38, 0)
2019-11-25 20:55:45.670 2099-2099/com.ktxsoftware.kha E/MediaPlayer: Error (-38,0)
```

**Comment by RobDangerous** [comment]
I have to rename the Non-Native-Android backend to something silly, I get confused every single time when somebody uses it.

**Comment by Huyston** [comment]
> On Android the issue is a bit different, it wasn't working because of a bug in my code.
Stop works but you can't play the audio again:

Just to be clear, this happens in both Android Native and Non Native Android backend.

Log from native Android backend:

```2019-11-30 18:43:35.142 22198-22225/com.ktxsoftware.kha E/MediaPlayerNative: Attempt to perform seekTo in wrong state: mPlayer=0x72da81f080, mCurrentState=64
2019-11-30 18:43:35.142 22198-22225/com.ktxsoftware.kha E/MediaPlayerNative: error (-38, 0)
2019-11-30 18:43:35.143 22198-22198/com.ktxsoftware.kha E/MediaPlayer: Error (-38,0)
2019-11-30 18:43:35.145 22198-22225/com.ktxsoftware.kha E/MediaPlayerNative: start called in state 0, mPlayer(0x72da81f080)
2019-11-30 18:43:35.145 22198-22225/com.ktxsoftware.kha E/MediaPlayerNative: error (-38, 0)
2019-11-30 18:43:35.146 22198-22198/com.ktxsoftware.kha E/MediaPlayer: Error (-38,0)```

**Comment by Huyston** [comment]
This happens because stop actually puts the media into a final state. A more intuitive way to handle stop is to pause it then rewind it. So that you can play it again in the future.

This is very simple to implement. I can make a PR if this is ok.

**Comment by Huyston** [comment]
These changes makes the API consistent across backends, because that is the same behavior of Android Native and HTML5.

**Comment by Huyston** [comment]
Changing the name as PR 1169 fix the android issue.
